### PR TITLE
cleanup, update packages, minor fixes

### PR DESCRIPTION
**Pull request by mgrybyk**
### repo changes

- fix eslint errors/warnings
- introduce github workflow with eslint (REQUIRES APPROVAL)
- use the latest version of nodejs 16
- allow building `headless` version without copying/replacing files (readme updated)
- update dependencies

### code changes

- replaced variable reassignments in some places
- fixed `updateWorker`
- filter out targets with missing/invalid method (no need in `algorithm === null` anymore)
- fixed minor typos 

### testing

I've tested the following to make sure my changes won't break anything 

- electron build/start works
- headless build/start works
- docker build and run works

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "shield",
-  "version": "0.0.18",
+  "version": "0.0.19",
```
Comment (opengs) [inline package.json:3]: dont change. this will create new version and run CI/CD :D

File: packageheadless.json [hunk 2]
```diff
@@ -1,68 +1,27 @@
...
-    "path": "^0.12.7",
-    "simple-ddos": "^2.0.4",
-    "superagent": "^7.1.1",
-    "vue": "^3.0.0",
-    "vue-i18n": "^9.0.0",
-    "vue-router": "^4.0.0",
-    "user-agents": "^1.0.943",
-    "vuex": "^4.0.1",
-    "uuid4": "^2.0.2",
-    "universal-analytics": "^0.5.3"
```
Comment (opengs) [inline packageheadless.json:32]: headless version probably will not build. at least several packages are required. try delete node_modules and install from packageheadless.json

File: src-worker/algorithms/group.ts [hunk 3]
```diff
@@ -13,9 +13,7 @@ export class AlgorithmGroup {
     this.algorithmMap.set('post', new Post(config, proxyPool))
   }
 
-  getByType (algorithmType: AlgorithmType) : Algorithm | null {
-    const alg = this.algorithmMap.get(algorithmType)
-    if (alg === undefined) return null
```
Comment (opengs) [inline src-worker/algorithms/group.ts:18]: this is for backward compatibility. map may not contain newer algorithm and `this.algorithmMap.get(algorithmType)` returns `Algorithm | undefined` . So there was null check added. Its at least must be changed to return type `Algorithm | undefined`

File: src-worker/external/targetsPool.ts [hunk 4]
```diff
@@ -63,17 +63,16 @@ export class TargetsPool {
...
     // Load proxyes from each resource
-    let loadedTargetsList = [] as Target[]
+    const loadedTargetsList = [] as Target[]
     for (const url of this.sources) {
       const targets = await this.loadFromURL(url)
       if (typeof targets === 'string') {
         console.warn(targets)
       } else {
-        loadedTargetsList = [...loadedTargetsList, ...targets]
+        loadedTargetsList.push(...targets.filter((t) => t.method === 'get' || t.method === 'post'))
```
Comment (opengs) [inline src-worker/external/targetsPool.ts:72]: new target types where added. please check 'udp_flood' and 'slowloris'

File: src-worker/planing/executor.ts [hunk 5]
```diff
@@ -58,17 +59,12 @@ export class Executor {
...
       // Wait for targets
-      await new Promise(resolve => setTimeout(resolve, 5000))
+      await sleep(5000)
       return
     }
 
     const algorithm = this.algorithmGroup.getByType(target.method)
 
-    if (algorithm === null) {
-      await new Promise(resolve => setTimeout(resolve, 50))
```
Comment (opengs) [inline src-worker/planing/executor.ts:68]: algorithm may be null for old version of the program

**Review by opengs — CHANGES_REQUESTED** [review]

File: packageheadless.json (hunk 2 above)
Comment (mgrybyk) [reply packageheadless.json:32]: you are right, missed it, messed up while switching between these files

File: src-worker/algorithms/group.ts (hunk 3 above)
Comment (mgrybyk) [reply src-worker/algorithms/group.ts:18]: what if we set method to `get` if it's not provided?

File: src-worker/planing/executor.ts (hunk 5 above)
Comment (mgrybyk) [reply src-worker/planing/executor.ts:68]: why would we wait for 50 ms in such a case? If the `algorithm` is `null` it won't ever change to something else

File: src-worker/external/targetsPool.ts (hunk 4 above)
Comment (mgrybyk) [reply src-worker/external/targetsPool.ts:72]: let me think how to deal with it

**Comment by mgrybyk** [comment]
Let me raise PR to 0.0.x

**Pull request closed without merging**